### PR TITLE
fix: ci-cd count per day in telemetry data

### LINKED ISSUE #4886: Bug: CI/CD count per day is not correct in telemetry data
### 📜 Description

CI/CD count in telemetry data is not correct.

### 👟 Reproduction steps

NA

### 👍 Expected behavior

Data should be the number of CI/CD triggered.

### 👎 Actual Behavior

The data getting pushed is the CI/CD count newly created in the last 24 hours.

### ☸ Kubernetes version

EKS 1.27

### Cloud provider

<details>
AWS
</details>

### 🌍 Browser

Chrome

### 🧱 Your Environment

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by komalreddy3**
# Description

Fixes #4886

Added fields to TelemetryEventDto

1. ciTriggeredPerDay ( count of ci pipelines triggered in last 24hrs)
2. cdTriggeredPerDay ( count of cd pipelines triggered in last 24hrs)
3. ciDeletedPerDay ( count of ci pipelines deleted in last 24hrs)
4. cdDeletedPerDay ( count of cd pipelines deleted in last 24hrs)
5. Rename ciCountPerDay to ciCreatedPerDay ( count of ci pipelines created in last 24hrs)
6. Rename cdCountPerDay to cdCreatedPerDay ( count of cd pipelines created in last 24hrs)

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: internal/sql/repository/pipelineConfig/CdWorfkflowRepository.go [hunk 1]
```diff
@@ -357,7 +357,15 @@ func (impl *CdWorkflowRepositoryImpl) FindLatestCdWorkflowByPipelineIdV2(pipelin
 	// TODO - Group By Environment And Pipeline will get latest pipeline from top
 	return cdWorkflow, err
 }
-
+func (impl *CdWorkflowRepositoryImpl) FindAllTriggeredWorkflowInLast24Hour() (cdWorkflowCount int, err error) {
+	var workflowCount int
+	query := "SELECT count(DISTINCT(cd_workflow_id)) FROM cd_workflow_runner WHERE started_on > now() - interval '1 day'"
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/CdWorfkflowRepository.go:362]: use go-pg to perform the same native query

Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/CdWorfkflowRepository.go:362]: cd_workflow_runner has 3 runner types: PRE/ POST/ DEPLOY. we'll fetch the deployments only I guess. verify it from Product.

File: internal/sql/repository/pipelineConfig/CiWorkflowRepository.go [hunk 2]
```diff
@@ -286,7 +287,16 @@ func (impl *CiWorkflowRepositoryImpl) FindLastTriggeredWorkflowByArtifactId(ciAr
 		Select()
 	return workflow, err
 }
+func (impl *CiWorkflowRepositoryImpl) FindAllTriggeredWorkflowInLast24Hour() (ciWorkflowCount int, err error) {
 
+	var workflowCount int
+	query := "SELECT count(DISTINCT(ci_pipeline_id)) FROM ci_workflow WHERE started_on > now() - interval '1 day'"
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/CiWorkflowRepository.go:293]: use go-pg to achieve the same query

**Review by Ash-exp — CHANGES_REQUESTED** [review]

**Review by Ash-exp — DISMISSED** [review]
LGTM

**Review by Ash-exp — APPROVED** [review]

File: internal/sql/repository/dockerRegistry/DockerArtifactStoreRepository.go [hunk 3]
```diff
@@ -169,7 +170,19 @@ func (impl DockerArtifactStoreRepositoryImpl) FindAll() ([]DockerArtifactStore,
 		Select()
 	return providers, err
 }
-
+func (impl DockerArtifactStoreRepositoryImpl) FindAllDockerArtifactCount() (int, error) {
+	dockerArtifactCount, err := impl.dbConnection.Model(&DockerArtifactStore{}).
+		Column("docker_artifact_store.*", "IpsConfig", "OCIRegistryConfig").
```
Comment (Ash-exp) [inline internal/sql/repository/dockerRegistry/DockerArtifactStoreRepository.go:175]: "IpsConfig", "OCIRegistryConfig" relations can be dropped

File: internal/sql/repository/pipelineConfig/CiPipelineRepository.go [hunk 4]
```diff
@@ -502,12 +503,19 @@ func (impl *CiPipelineRepositoryImpl) FinDByParentCiPipelineAndAppId(parentCiPip
 	return ciPipelines, err
 }
 
-func (impl *CiPipelineRepositoryImpl) FindAllPipelineInLast24Hour() (pipelines []*CiPipeline, err error) {
-	err = impl.dbConnection.Model(&pipelines).
+func (impl *CiPipelineRepositoryImpl) FindAllPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&CiPipeline{}).
 		Column("ci_pipeline.*").
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/CiPipelineRepository.go:508]: do we require all columns, ci_pipeline.*. I guess just ci_pipeline.id will do

File: internal/sql/repository/pipelineConfig/CiPipelineRepository.go [hunk 5]
```diff
@@ -502,12 +503,19 @@ func (impl *CiPipelineRepositoryImpl) FinDByParentCiPipelineAndAppId(parentCiPip
...
 		Where("created_on > ?", time.Now().AddDate(0, 0, -1)).
-		Select()
-	return pipelines, err
+		Count()
+	return pipelineCount, err
+}
+func (impl *CiPipelineRepositoryImpl) FindAllDeletedPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&CiPipeline{}).
+		Column("ci_pipeline.*").
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/CiPipelineRepository.go:515]: same here

File: internal/sql/repository/pipelineConfig/PipelineRepository.go [hunk 6]
```diff
@@ -447,12 +448,19 @@ func (impl PipelineRepositoryImpl) FindByPipelineTriggerGitHash(gitHash string)
 	return pipelines, err
 }
 
-func (impl PipelineRepositoryImpl) FindAllPipelineInLast24Hour() (pipelines []*Pipeline, err error) {
-	err = impl.dbConnection.Model(&pipelines).
+func (impl PipelineRepositoryImpl) FindAllPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&Pipeline{}).
 		Column("pipeline.*").
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/PipelineRepository.go:453]: same here

File: internal/sql/repository/pipelineConfig/PipelineRepository.go [hunk 7]
```diff
@@ -447,12 +448,19 @@ func (impl PipelineRepositoryImpl) FindByPipelineTriggerGitHash(gitHash string)
...
 		Where("created_on > ?", time.Now().AddDate(0, 0, -1)).
-		Select()
-	return pipelines, err
+		Count()
+	return pipelineCount, err
+}
+func (impl PipelineRepositoryImpl) FindAllDeletedPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&Pipeline{}).
+		Column("pipeline.*").
```
Comment (Ash-exp) [inline internal/sql/repository/pipelineConfig/PipelineRepository.go:460]: here

File: pkg/cluster/repository/EnvironmentRepository.go [hunk 8]
```diff
@@ -256,6 +257,15 @@ func (repositoryImpl EnvironmentRepositoryImpl) FindAllActive() ([]*Environment,
 		Select()
 	return mappings, err
 }
+func (repositoryImpl EnvironmentRepositoryImpl) FindAllActiveEnvironmentCount() (int, error) {
+	cnt, err := repositoryImpl.
+		dbConnection.Model(&Environment{}).
+		Where("environment.active = ?", true).
+		Column("environment.*", "Cluster").
```
Comment (Ash-exp) [inline pkg/cluster/repository/EnvironmentRepository.go:264]: drop the cluster join

**Review by Ash-exp — CHANGES_REQUESTED** [review]

File: internal/sql/repository/pipelineConfig/CiPipelineRepository.go [hunk 9]
```diff
@@ -502,12 +503,19 @@ func (impl *CiPipelineRepositoryImpl) FinDByParentCiPipelineAndAppId(parentCiPip
 	return ciPipelines, err
 }
 
-func (impl *CiPipelineRepositoryImpl) FindAllPipelineInLast24Hour() (pipelines []*CiPipeline, err error) {
-	err = impl.dbConnection.Model(&pipelines).
-		Column("ci_pipeline.*").
+func (impl *CiPipelineRepositoryImpl) FindAllPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&CiPipeline{}).
+		Column("ci_pipeline.id").
```
Comment (nishant-d) [inline internal/sql/repository/pipelineConfig/CiPipelineRepository.go:508]: do we need `Column("ci_pipeline.id").` here

File: internal/sql/repository/pipelineConfig/CiPipelineRepository.go [hunk 10]
```diff
@@ -502,12 +503,19 @@ func (impl *CiPipelineRepositoryImpl) FinDByParentCiPipelineAndAppId(parentCiPip
...
 		Where("created_on > ?", time.Now().AddDate(0, 0, -1)).
-		Select()
-	return pipelines, err
+		Count()
+	return pipelineCount, err
+}
+func (impl *CiPipelineRepositoryImpl) FindAllDeletedPipelineCountInLast24Hour() (pipelineCount int, err error) {
+	pipelineCount, err = impl.dbConnection.Model(&CiPipeline{}).
+		Column("ci_pipeline.id").
```
Comment (nishant-d) [inline internal/sql/repository/pipelineConfig/CiPipelineRepository.go:515]: do we need `Column("ci_pipeline.id").` here

File: internal/sql/repository/GitOpsConfigRepository.go [hunk 11]
```diff
@@ -91,6 +92,10 @@ func (impl *GitOpsConfigRepositoryImpl) GetAllGitOpsConfig() ([]*GitOpsConfig, e
 	err := impl.dbConnection.Model(&userModel).Order("updated_on desc").Select()
 	return userModel, err
 }
+func (impl *GitOpsConfigRepositoryImpl) GetAllGitOpsConfigCount() (int, error) {
+	cnt, err := impl.dbConnection.Model(&GitOpsConfig{}).Order("updated_on desc").Count()
```
Comment (nishant-d) [inline internal/sql/repository/GitOpsConfigRepository.go:96]: .Order("updated_on desc")

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Review by nishant-d — APPROVED** [review]

**Review by Ash-exp — APPROVED** [review]

**Pull request merged**